testing/tests/basic.rs

```diff
@@ -5,74 +5,150 @@
+    let dead_nodes = vec![ChangeNode {
+        idx: 10,
+        updown: UpDown::Down,
+    }];
+
+    metadata.spinning_properties = SpinningTaskDescription {
+        node_changes: vec![(Duration::new(4, 0), dead_nodes)],
+    };
     metadata
-        .gen_launcher::<hotshot_testing::node_types::SequencingTestTypes, hotshot_testing::node_types::SequencingMemoryImpl>()
+        .gen_launcher::<SequencingTestTypes, SequencingMemoryImpl>()
         .launch()
         .run_test()
         .await;
 }
 
+/// Test f/2 nodes leaving the network.
 #[cfg(test)]
 #[cfg_attr(
     feature = "tokio-executor",
     tokio::test(flavor = "multi_thread", worker_threads = 2)
 )]
 #[cfg_attr(feature = "async-std-executor", async_std::test)]
-// TODO Keyao (testing harness) Remove `#[ignore]` after making the test pass. It's currently
-// hanging.
-#[ignore]
-async fn test_with_failures() {
+async fn test_with_failures_half_f() {
     use std::time::Duration;
 
     use hotshot_testing::{
-        completion_task::TimeBasedCompletionTaskDescription, spinning_task::SpinningTaskDescription,
+        node_types::{SequencingMemoryImpl, SequencingTestTypes},
+        spinning_task::{ChangeNode, SpinningTaskDescription, UpDown},
+        test_builder::TestMetadata,
     };
 
     async_compatibility_layer::logging::setup_logging();
     async_compatibility_layer::logging::setup_backtrace();
-    let mut metadata = hotshot_testing::test_builder::TestMetadata {
-        total_nodes: 20,
-        start_nodes: 20,
-        num_bootstrap_nodes: 20,
-        da_committee_size: 20,
-        completion_task_description: hotshot_testing::completion_task::CompletionTaskDescription::TimeBasedCompletionTaskBuilder(TimeBasedCompletionTaskDescription{duration: Duration::new(120, 0)}),
-        // overall_safety_properties: OverallSafetyPropertiesDescription {
-        //     threshold_calculator: std::sync::Arc::new(|_, _| {10}),
-        //     ..Default::default()
-        // },
-        ..hotshot_testing::test_builder::TestMetadata::default()
+    let mut metadata = TestMetadata::default_more_nodes_less_success();
+    let dead_nodes = vec![
+        ChangeNode {
+            idx: 5,
+            updown: UpDown::Down,
+        },
+        ChangeNode {
+            idx: 10,
+            updown: UpDown::Down,
+        },
+        ChangeNode {
+            idx: 15,
+            updown: UpDown::Down,
+        },
+    ];
+
+    metadata.spinning_properties = SpinningTaskDescription {
+        node_changes: vec![(Duration::new(4, 0), dead_nodes)],
+    };
+    metadata
+        .gen_launcher::<SequencingTestTypes, SequencingMemoryImpl>()
+        .launch()
+        .run_test()
+        .await;
+}
+
+/// Test f nodes leaving the network.
+#[cfg(test)]
+#[cfg_attr(
+    feature = "tokio-executor",
+    tokio::test(flavor = "multi_thread", worker_threads = 2)
+)]
+#[cfg_attr(feature = "async-std-executor", async_std::test)]
+async fn test_with_failures_f() {
```

Comment:
Nit: Maybe we could add a quick comment that these failures purposefully don't include DA nodes.  I feel like we may forget this in the future when we switch to a rotating DA committee, and then we'll wonder why these tests fail. :)

Reply:
Oh, the failures actually include the DA nodes. The DA committee is the entire network for this test, so those 6 nodes to be killed are also DA nodes.

Reply:
Now failures only include DA nodes, so I added a comment.